File: FinGenieAI/src/main/java/com/fingenie/service/AccountService.java
package com.fingenie.service;
 
import java.util.List;
import com.fingenie.dto.AccountDTO;
 
public interface AccountService {
 
    AccountDTO createAccount(AccountDTO accountDTO);
 
    AccountDTO getAccountById(Long accountId);
 
    List<AccountDTO> getAllAccounts();
    
    AccountDTO updateAccount(Long accountId,AccountDTO accountDTO);
 
    void deleteAccount(Long accountId);
}

File: frontend/fingenie-frontend/src/services/accountService.js
import api from "./api";

export const getAllAccounts = async () => {
  const response = await api.get("/accounts");
  return response.data;
};

export const getAccountById = async (id) => {
  const response = await api.get(`/accounts/${id}`);
  return response.data;
};

export const createAccount = async (accountData) => {
  const response = await api.post(
    "/accounts",
    accountData
  );

  return response.data;
};

export const updateAccount = async (
  id,
  accountData
) => {
  const response = await api.put(
    `/accounts/${id}`,
    accountData
  );

  return response.data;
};

export const deleteAccount = async (id) => {
  const response = await api.delete(
    `/accounts/${id}`
  );

  return response.data;
};